src/client/datascience/notebook/vscodeNotebookController.ts

```diff
@@ -167,6 +171,8 @@ export class VSCodeNotebookController implements Disposable {
         if (!this.workspace.isTrusted) {
             return;
         }
+        initializeNotebookTelemetryBasedOnUserAction(notebook.uri, this.connection);
+        sendKernelTelemetryEvent(notebook.uri, Telemetry.ExecuteCell);
```

Comment:
User ran a cell, this is called by VS Code when user runs cells via  VS Code built in commands.
Doing this earlier guarantees the fact that we'll properly track this even as a deliberate action by user to start a notebook/run a cell.

Reply:
Other parts of the code can get executed via export/interactive window, etc.